core/controllers/contributor_dashboard.py

```diff
@@ -361,54 +360,34 @@ def _get_reviewable_exploration_opportunity_summaries(
         Returns:
             list(ExplorationOpportunitySummary). A list of the matching
             exploration opportunity summaries.
-
-        Raises:
-            Exception. No exploration_id found for the node_id.
         """
-        # 1. Fetch the eligible topics.
-        # 2. Fetch the stories for the topics.
-        # 3. Get the reviewable translation suggestion target IDs for the user.
-        # 4. Get story exploration nodes in order, filtering for explorations
-        # that have in review translation suggestions.
-        if topic_name is None:
-            topics = topic_fetchers.get_all_topics()
+        # 1. Fetch all the exploration IDs of published chapters in published
+        #    stories under the topic(s).
+        # 2. Fetch each reviewable translation suggestion to be available to the
+        #    translation reviewer.
+        # 3. Filter exploration ids. Keep those that are the target of any
+        #    fetched reviewable translation suggestions.
+        # 4. Fetch all exploration opportunity summaries for the kept
+        #    explorations.
+        # 5. If there's a pinned lesson, move the pinned lesson's summary to
+        #    the top.
```

Comment:
I found that I needed to read this set of steps multiple times to understand what was going on. Maybe we can streamline it a bit. E.g.:

1. Fetch the IDs of all published explorations in the topic's published stories.
2. Fetch all suggestions that the user can review, and collect their exploration IDs. Filter these IDs to only include published explorations in published stories (see step 1).
3. Fetch all exploration opportunity summaries for the resulting set of explorations.
4. Moved any pinned summaries to the top.